src/store/reducers/todos.js

```diff
@@ -0,0 +1,28 @@
+import {
+  ADD_TODO,
+  DELETE_TODO,
+  TOGGLE_TODO,
+} from "../actionsTypes/actionTypes";
+import { initialState } from "../initialState";
+
+export const reducer = (state = initialState, action) => {
+  switch (action.type) {
+    case ADD_TODO: {
+      const todo = {
+        id: new Date(),
+        message: action.payload.message,
+        isCompleted: false,
+      };
+      state.push(todo);
```

Comment:
Reducer is a pure function. You should not mutate the state calling push(), pop(), etc
You should create a new store copy on every change